python/cudf/cudf/io/parquet.py

```diff
@@ -834,6 +842,12 @@ def _parquet_to_frame(
                 key_paths,
                 *args,
                 row_groups=key_row_groups,
+                # TODO: is this still right?
+                # Also, do we still care?
+                # partition_keys uses pyarrow dataset
+                # (which we can't use anymore after pyarrow is gone)
+                nrows=nrows,
+                skip_rows=skip_rows,
```

Comment:
No, I think this is wrong. These `dfs` are concatenated vertically, so after reading each df, one should do:
```
nrows = max(nrows - len(df), 0)
```

Updating `skip_rows` is more complicated because if you skipped all the rows in a file then you can't know if you should reduce `skip_rows` to zero or to `skip_rows - num_rows_in_file`.

Reply:
Multiple files are, I believe, an implementation detail.

Reply:
But let's cc @rjzamora to confirm

Reply:
It seems a feels a bit ugly/inefficient to use `nrows` and/or `skiprows` when we are reading from a partitioned dataset, but you are right that we wouldn't want to pass these parameters "as provided" to `_read_parquet`.

The "efficient" solution would probably have us read the row-count metadata for each element of `key_paths` up front. This way we could avoid reading unnecessary files/rows altogether. Of course, if the user passes in a filter, we would need to read all data with the filter and perform the row-trimming after the fact.

>which we can't use anymore after pyarrow is gone

My understanding is that the pyarrow-removal effort only applies to the cython/c++ level. We are still allowed to depend on pyarrow at the python level (removing pyarrow would be a nightmare for dask-cudf at this point).

Reply:
>At any rate, I think I might punt on this for now (since it doesn't look like making nrows/skiprows work for a partitioned dataset is high priority), and the rest of this PR is blocking nrows/skiprows support in the polars executor.

Yes - I don't see any reason to spend time optimizing nrows/skiprows for partitioned datasets. We can always revisit if necessary.